### PR TITLE
[HS-1245681] Hiding child question blocks

### PR DESCRIPTION
**Pull request by caleballdrin**
https://secure.helpscout.net/conversation/2738597988/1245681?folderId=7669074

Hide questions if it has a rule and the question that the rule is based on is hidden.

### CODE REVIEW

File: test/spec/directives/block.spec.js [hunk 1]
```diff
@@ -47,6 +47,16 @@ describe('Directive: blocks', () => {
...
+      it('is false when currentRegistration is null', () => {
+        $scope.currentRegistration = null;
+        globalUserSpy.and.returnValue(null);
+
+        $compile('<name-question></name-question>')($scope);
+        $scope.$digest();
+
+        expect($scope.lockedStaffProfileBlock).toBe(false);
+      });
+
```
Comment (caleballdrin) [inline test/spec/directives/block.spec.js:59]: @canac I'm not sure how these other 2 commits got in here. Do you think I just never pushed them from a long time ago?

File: test/spec/testData.spec.js [hunk 2]
```diff
@@ -411,6 +411,157 @@ angular.module('confRegistrationWebApp').service('testData', function () {
           },
         ],
       },
+      {
```
Comment (caleballdrin) [inline test/spec/testData.spec.js:414]: @canac Do you think I need to add `conference` to line 9 on [pageRegistration.html](https://github.com/CruGlobal/conf-registration-web/blob/44447f09d06ece3cceaccf3e00bac31c2e92adff/app/views/components/pageRegistration.html) ng-`show="blockVisible(block)"`

Comment (canac) [reply test/spec/testData.spec.js:414]: I don't think so. I'm not fully sure how, but it appears that `pageRegistration.html` is connected to the `blockRegistration` controller, which sets up `$scope.blockVisible`. That's the function that the template is calling.

**Comment by wrandall22** [comment]
For PRs that are from HS tickets, I prefer we use the `HS-123456` instead of `[No-Jira]` as a prefix in the title. This is more explanatory.

File: app/scripts/services/validateRegistrant.js [hunk 3]
```diff
@@ -73,6 +82,19 @@ angular
...
+            if (
+              conference &&
+              !blockVisibleRuleCheck(
+                parentBlock,
+                registrant,
+                ruleType,
+                conference,
+              )
+            ) {
+              return false;
```
Comment (canac) [inline app/scripts/services/validateRegistrant.js:96]: Returning false makes `_.forEach` stop iterating.

```suggestion
              return;
```

**Review by canac — APPROVED** [review]
Let's ship it! 🚀

**Pull request merged**